src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs

```diff
@@ -551,6 +561,85 @@ public void create_fieldbuilder_with_inferred_graph_type()
         type.Fields.Find("field13").ShouldNotBeNull().Type.ShouldBe(typeof(NonNullGraphType<ListGraphType<GraphQLClrOutputTypeReference<string>>>));
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void infer_nullability_from_nrt(bool infer)
+    {
+        bool old = GlobalSwitches.InferFieldNullabilityFromNRTAnnotations;
+
+        try
+        {
+            GlobalSwitches.InferFieldNullabilityFromNRTAnnotations = infer;
+            var schema = new Schema();
+            var type = new ComplexType<NrtTest>();
+            _ = type.Field(d => d.Str1);
+            _ = type.Field(d => d.Str2);
+            _ = type.Field(d => d.Str3);
+            _ = type.Field(d => d.List1);
+            _ = type.Field(d => d.List2);
+            _ = type.Field(d => d.List3);
+            _ = type.Field("concatStr", d => d.Str1 + d.List2);
+            _ = type.Field(d => d.Int1);
+            ShouldlyExtensions.ShouldThrowWhen<ArgumentException>(!infer, () => type.Field(d => d.Int2));
+
+            schema.Query = type;
+            schema.Initialize();
+
+            var field = type.Fields.FirstOrDefault(f => f.Name == "str1").ShouldNotBeNull();
+            field.Type.ShouldBe(typeof(NonNullGraphType<StringGraphType>));
+
+            field = type.Fields.FirstOrDefault(f => f.Name == "str2").ShouldNotBeNull();
+            field.Type.ShouldBeWhen(infer, typeof(StringGraphType), typeof(NonNullGraphType<StringGraphType>));
```

Comment:
I think without the ability to gradually migrate the schema type-by-type, this feature will stay disabled forever for large schemas...

Reply:
I think you are right. If NRT is enabled and user has a `string?`, for example, they probably use `nullable: true` to set the correct field nullability. Enabling this feature will change nothing in this case. If they have a non-nullable `string`, then nothing changes again. And if they have NRT disabled, why would they want to enable this feature?

Reply:
Yup. And if they wanted all nullable fields, as some devs do to prevent null bubbling, again they would have nullable: true so the feature can be safely enabled.